doc/source/development/roadmap.rst

```diff
@@ -0,0 +1,145 @@
+of these goals may be hastened with dedicated funding.
+
+Extensibility
+-------------
+
+Pandas Extension Arrays provide 3rd party libraries the ability to
+extend pandas' supported types. In theory, these 3rd party types can do
+everything one of pandas. In practice, many places in pandas will unintentionally
+convert the ExtensionArray to a NumPy array of Python objects, causing
+performance issues and the loss of type information. These problems are especially
+pronounced for nested data.
+
+We'd like to improve the handling of extension arrays throughout the library,
+making their behavior more consistent with the handling of NumPy arrays.
+
+String Dtype
+------------
+
+Currently, pandas stores text data in an ``object`` -dtype NumPy array.
+Each array stores Python strings. While pragmatic, since we rely on NumPy
+for storage and Python for string operations, this is memory inefficient
+and slow. We'd like to provide a native string type for pandas.
+
+The most obvious alternative is Apache Arrow. Currently, Arrow provides
+storage for string data. We can work with the Arrow developers to implement
+the operations needed for pandas users (for example, ``Series.str.upper``).
+These operations could be implemented in Numba (
+as prototyped in `Fletcher <https://github.com/xhochy/fletcher>`__)
+or in the Apache Arrow C++ library.
+
+Apache Arrow Interoperability
+-----------------------------
+
+`Apache Arrow <https://arrow.apache.org>`__ is a cross-language development
+platform for in-memory data. The Arrow logical types are closely aligned with
+typical pandas use cases (for example, support for nullable integers).
+
+We'd like have a pandas DataFrame be backed by Arrow memory and data types
+by default. This should simplify pandas internals and ensure more consistent
+handling of data types through operations.
+
+Block Manager Rewrite
+---------------------
+
+We'd like to replace pandas current internal data structures (a collection of
+1 or 2-D arrays) with a simpler collection of 1-D arrays.
+
+Pandas internal data model is quite complex. A DataFrame is made up of
+one or more 2-dimension "blocks", with one or more blocks per dtype. This
+collection of 2-D arrays is managed by the BlockManager.
+
+The primary benefit of the BlockManager is improved performance on certain
+operations (construction from a 2D array, binary operations, reductions across the columns),
+especially for wide DataFrames. However, the BlockManager substantially increases the
+complexity and maintenance burden of pandas'.
+
+By replacing the BlockManager we hope to achieve
+
+* Substantially simpler code
+* Easier extensibility with new logical types
+* Better user control over memory use and layout
+* Improved microperformance
+* Option to provide a C / Cython API to pandas' internals
+
+See `these design documents <https://dev.pandas.io/pandas2/internal-architecture.html#removal-of-blockmanager-new-dataframe-internals>`__
+for more.
```

Comment:
I _think_ the proposal in that document is mostly unrelated to our last discussions (e.g. 1D storage, or relying of Arrow)?

Reply:
I'm not sure I understand your comment.

The BlockManager rewrite (C/ Cythonized data structure holding 1D columns) and Arrow "by default" are indeed two different items. This one is about the BlockManager rewrite.

Reply:
OK, Arrow was OT. My point was just that "C/ Cythonized data structure holding 1D columns" does not seem to be related to the content of that page. But I didn't understand everything in that page, so I might be wrong.

Reply:
The document I linked to discusses some of the shortcomings of current implementation, and proposes a replacement Cythonized container of 1D arrays (libpandas). I think they're related.